### PR TITLE
For #5764: change download path

### PR DESCRIPTION
**Pull request by eliserichards**
For #5764 
Depends on AC pr: https://github.com/mozilla-mobile/android-components/pull/8194

todo:
 - [ ] land AC pr - in progress
 - [x] what is the correct way of pulling the settings from `core.engine` when initializing Fenix settings and when saving a new value?
 - [x] allow user to choose between local storage `Environment.DIRECTORY_DOWNLOADS` and an SD card if present
   - we don't want to let them save anywhere because 1. we'd have to validate the path, and 2. Android 11 will introduce [scoped storage](https://developer.android.com/preview/privacy/storage)

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt [hunk 1]
```diff
@@ -291,8 +291,13 @@ class BrowserFragment : BaseBrowserFragment(), UserInteractionHandler {
...
             view,
-            FenixSnackbarDelegate(view)
+            FenixSnackbarDelegate(view),
+            getDefaultDownloadPath()
         ) + ContextMenuCandidate.createOpenInExternalAppCandidate(requireContext(),
             contextMenuCandidateAppLinksUseCases)
     }
+
+    private fun getDefaultDownloadPath(): String {
+        return requireActivity().settings().downloadPath
```
Comment (Amejia481) [inline app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt:301]: It this getting the latest updated value? I was experiencing issues using the `setting` class see https://github.com/mozilla-mobile/fenix/issues/13378. I was forced to use : `PreferenceManager.getDefaultSharedPreferences(context)` to get the latest value

Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt:301]: I was thinking that maybe `getDefaultDownloadDirectoy` must clear, I think we should do the same on ac part :)

**Comment by eliserichards** [comment]
Android 11 is going to introduce some issues in the way we are handling downloads. Currently--in Android 10 and below--an app can have access to any & all directories given the read/write permission. This means that there's no differentiation between public directories (like `/Download`) or private ones. Starting in Android 11, storage will be restricted; apps will only have access to public directories and their own directories, and permissions will be required for access to any others. 

We are seeing this functionality in Opera, where

**Comment by eliserichards** [comment]
Closing for now. See comment: https://github.com/mozilla-mobile/fenix/issues/5764#issuecomment-683879718

**Pull request closed without merging**